### DISCUSSION TITLE
Queue Size limit and Close Queue

### DISCUSSION THREAD

**Discussion by lgcCerti** [taskforcesh/bullmq#2783] [Q&A]
Hello,

I've been looking for queue messaging systems for NodeJS and came across BullMQ for interprocess communication.

Currently in BullMQ, I've only found ways to add into the queue and consume it. 

There are moments when I would like for the Queue to be frozen and not receive jobs. In the documentation I'm allowed to `pause()` the queue, but this reads that it merely pauses the jobs from being processed and they will still queue up indefinitely.

Are there no mechanisms that allow for limiting queue size, and by extension preventing a queue from receiving new jobs? Otherwise, I (and others) would need to work my code around BullMQ (were jobs added with a "not running" signal? discard the job data).

Thanks.

**Comment by manast** [comment]
Not sure I understand this question. Basically if you do not want to add more jobs to the queue just don't do it :) Probably I am missing something here...

**Reply by lgcCerti** [reply]
> Not sure I understand this question. Basically if you do not want to add more jobs to the queue just don't do it :) Probably I am missing something here...

One may no longer have a direct reference to the routines to stop inserters manually, nor would they want to do this. The inserters may instead handle an error offered by the API.

Likewise for .pause() and .obliterate() where one wishes for workers to pause or to remove all jobs, I'd like the queue to merely stop receiving jobs until given the signal, abdicating either. 

Otherwise, I'm effectively calling `obliterate()` then resuming the queue whenever to simulate this. Which is frankly unreadable. Another way would be to wrap the BullMQ API with another API that only lets things added to Queue when conditions are met. Which may be worse.
